source/Related-Projects/Ogma.rst

```diff
@@ -0,0 +1,31 @@
+Ogma
+====
+
+NASA's `Ogma <https://github.com/nasa/ogma>`_ is an open source tool to generate runtime monitoring applications.
+
+The purpose of Runtime Verification is to monitor a system at runtime and detect anomalies or violations of expected properties. RV concerns itself with detecting the violations; it is the user's responsibility to determine how to address them.
+
+Ogma facilitates the generation of correct monitors by translating high-level languages into self-contained monitoring applications.
+
+Ogma supports a number of source languages, including `Copilot <https://github.com/copilot-language/copilot>`_, `FRET <FRET>` and Lustre. The applications can be generated for the Robot Operating System (SpaceROS) and `NASA's Core Flight System (cFS) <https://github.com/nasa/cFS/>`_, among others.
+
+Why use Ogma?
+-------------
+
+Runtime Verification is crucial especially for systems that are not amenable to full formal verification, or when testing all possible cases is not viable. It can also prove exceptionally useful to detect anomalies in the environment.
```

Comment:
I'm not quite clear about "anomalies in the environment." Is this referring to the environment in which the robot is operating? Or, perhaps the run-time environment of the software. Can you clarify, please?

Reply:
I believe this is now resolved.